solc/CommandLineInterface.cpp

```diff
@@ -1202,6 +1220,17 @@ void CommandLineInterface::outputCompilationResults()
 	handleAst(g_argAstJson);
 	handleAst(g_argAstCompactJson);
 
+	if(checkOutputComponent(m_args))
+	{
+		if(!m_args.count(g_argGas))
```

Comment:
I don't really like how this is a special case. Perhaps just call `needsHumanTargetedStdout' instead, even if it does more?

Reply:
Just to clarify myself are you asking to change `checkOutputComponent` to `needsHumanTargetedStdout`.

Reply:
or you are concerned about inner "if" condition

Reply:
Please have a look at https://gist.github.com/anurag-git/473ffe59df644e885e4be224ffce3b1d for design and tests performed.

Reply:
only the inner condition.